boards/veer_el2_sim/src/io.rs

```diff
@@ -1,48 +1,28 @@
 // Licensed under the Apache License, Version 2.0 or the MIT License.
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 // Copyright Tock Contributors 2022.
+// Copyright (c) 2024 Antmicro <www.antmicro.com>
 
-use core::fmt::Write;
 use core::panic::PanicInfo;
 use core::ptr::write_volatile;
-use core::str;
+use core::ptr::{addr_of, addr_of_mut};
 use kernel::debug;
-use kernel::debug::IoWrite;
+use veer_el2::io::Writer;
 
 use crate::CHIP;
 use crate::PROCESSES;
 use crate::PROCESS_PRINTER;
 
-struct Writer {}
-
 static mut WRITER: Writer = Writer {};
 
-impl Write for Writer {
-    fn write_str(&mut self, s: &str) -> ::core::fmt::Result {
-        self.write(s.as_bytes());
-        Ok(())
-    }
-}
-
-impl IoWrite for Writer {
-    fn write(&mut self, buf: &[u8]) -> usize {
-        for b in buf {
-            // Print to a special address for simulation output
-            unsafe {
-                write_volatile(0x8000_1008 as *mut u8, *b);
-            }
-        }
-        buf.len()
-    }
-}
```

Comment:
We will need a panic writer to remain implemented

Reply:
I believe the panic handler remains here. The Writer struct is changed because a single interface is used for both UART and panic.